.clang-tidy

```diff
@@ -0,0 +1,28 @@
+---
+Checks:          '*'
```

Comment:
Using `*` here is very aggressive. But it does not reveal any more than a few problems inside node-cpp-skel and they are easily fixed. So I think starting with this is ideal. Downstream users of node-cpp-skel will be able to change this to disable some checks (based on https://clang.llvm.org/extra/clang-tidy/) but opt-ing into all of them seems wisest to me to help keep code robust from the start.